kmapper/kmapper.py

```diff
@@ -735,25 +742,38 @@ def visualize(
         if colorscale is None:
             colorscale = colorscale_default
 
-        # TODO:
-        #   - Make color functions more intuitive. How do they even work?
-        #   - Allow multiple color functions that can be toggled on and off.
+        if X_names is None:
+            X_names = []
+
+        if lens_names is None:
+            lens_names = []
 
         if not len(graph["nodes"]) > 0:
             raise Exception(
                 "Visualize requires a mapper with more than 0 nodes. \nIt is possible that the constructed mapper could have been constructed with bad parameters. This can occasionally happens when using the default clustering algorithm. Try changing `eps` or `min_samples` in the DBSCAN clustering algorithm."
             )
 
-        # Find the module absolute path and locate templates
-        module_root = os.path.join(os.path.dirname(__file__), "templates")
-        env = Environment(loader=FileSystemLoader(module_root))
-        color_values = init_color_values(graph, color_values)
+        if isinstance(color_function_name, str):
+            color_function_name = [color_function_name]
 
-        if X_names is None:
-            X_names = []
+        # test whether we have a color_function_names for each color_value vector
+        if color_values is not None and color_function_name is not None:
+            num_color_value_vectors = color_values.shape[1]
+            num_color_function_names = len(color_function_name)
+            if num_color_value_vectors != num_color_function_names:
+                raise Exception('{} `color_function_names` values found, but {} columns found in color_values. Must be equal.'.format(num_color_function_names, num_color_value_vectors))
```

Comment:
I noticed the library still says it supports 2.7.  We should probably update this to be >3.6 only.  If we do that, you can use f-strings.

Reply:
Cool I'll look into that. And yeah, we say we support some really old python versions. I ❤️ only supporting >=3.6

Reply:
haven't done f-string yet. future PR?